setup.cfg

```diff
@@ -100,10 +100,19 @@ doctest_plus = enabled
 text_file_format = rst
 open_files_ignore = "astropy.log" "/etc/hosts" "*.ttf"
 remote_data_strict = true
-addopts = -p no:warnings --doctest-rst
+addopts = --doctest-rst
 asdf_schema_root = astropy/io/misc/asdf/data/schemas
 xfail_strict = true
 qt_no_exception_capture = 1
+filterwarnings =
+    error
+    ignore:unclosed file:ResourceWarning
+    ignore:unclosed <socket:ResourceWarning
+    ignore:unclosed <ssl.SSLSocket:ResourceWarning
+    ignore:The timefunc function is deprecated
+    ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:Remote data cache could not be accessed due to NotADirectoryError
+    ignore:.*Cache directory cannot be read or created
```

Comment:
@aarchiba , some remote data tests (both in test files and doctest) still emit these two warnings, so I ignore them for now. We can revisit after 4.0.